packages/portalnetwork/src/client/routingTable.ts

```diff
@@ -75,16 +79,39 @@ export class PortalNetworkRoutingTable extends KademliaRoutingTable {
   }
 
   /**
-   * Remove a node from the routing table
+   * Evict a node from the routing table and ignore
    * @param nodeId nodeId of peer to be evicted
    */
   public evictNode = (nodeId: NodeId) => {
     this.logger?.extend('EVICT')(nodeId)
-    const enr = this.getValue(nodeId)
+    let enr: ENR | undefined = this.getValue(nodeId)
+    this.ignoreNode(nodeId)
+    if (enr) {
+      enr = this.removeById(nodeId)?.value
+    }
+    if (enr) {
+      enr = this.remove(enr)?.value
+    }
     this.radiusMap.delete(nodeId)
     this.gossipMap.delete(nodeId)
-    if (enr) {
-      this.remove(enr)
+    this.strikes.delete(nodeId)
+  }
+
+  // Add node to ignored list for 2 minutes and then delete from ignored list
+
+  private ignoreNode = (nodeId: NodeId) => {
```

Comment:
What do you think about having a single interval that checks the ignore list periodically and clears out ones that have passed a 2 minute mark rather than setting up a bunch of individual timeouts?  If we amass a large routing table, this could be a lot of open listeners.  We could even bake it into the bucket refresh listener, kind of a global table refresher method.  You could have the ignore list be a map or a sorted list of nodeIds -> timestamp and just prune the list for all timestamps greater than `Date.now() - 120000`.  That way, we don't have to worry about lots of dangling listeners in tests or elsewhere.

Reply:
> You could have the ignore list be a map or a sorted list of nodeIds -> timestamp and just prune the list for all timestamps greater than Date.now() - 120000. That way, we don't have to worry about lots of dangling listeners in tests or elsewhere.

Clever. I'll do that.

Reply:
I think a slightly better way to do this would be to just create the ignore list as an array of tuples where we just push new tuples of `[timestamp, nodeId]` onto the end.  Then, we just do `splitIndex = ignoreList.findIndex((entry) => entry[0] > Date.now()-120000)` and then do `ignoreList = `ignoreList.slice(splitIndex)`.  What you're doing now is iterating over the entire ignoreList every time we do the bucket refresh but we should only need to find the first element with a timestamp < than 2 minutes ago and remove all items on the list prior to that.  Does that make sense?

Reply:
:+1: 
That works for me.  Made it so.